### PR TITLE
Code change for Article page

### PR DESCRIPTION
**Pull request by nickhsine**
1. change RESTful api on FE server(3030 port)
  - change endpoint 'articles' to 'posts'
  - Update api/api.js
  - add api/utils/url.js, this is for choosing right actions for
    corresponding urls
2. Update src/actions/{articles.js.article.js,tags}
  - replace 'authors' by more embedded entities, 'writters',
    'photographers', 'designers' and 'engineers'
  - update unit test
3. Add container for dealing with redux state
  - src/containers/Article.js is the entry of rendering article page
  - add components, src/components/article/HeadingAuthor.js and
    src/components/Author.js
4. Add routing rule for article page
5. add src/utils/denormalize-articles for composing complete article
entities
6. use 'react-hot-loader' for development
7. Code change for src/containers/Home.js for using new RESTful api response.

@hcchien @garfieldduck

### CODE REVIEW

File: src/actions/articles.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ function buildQueryURL(tags, count, page) {
   query.max_results = count || 10
   query.page = page || 0
   query.sort = '-publishedDate'
-  query.embedded = JSON.stringify( { authors: 1, tags:1, categories:1 } )
+  query.embedded = JSON.stringify({ writters: 1, photographers: 1, designers: 1, engineers:1, tags:1 })
```
Comment (hcchien) [inline src/actions/articles.js]: I guess you have to include the 'heroImage: 1'

File: src/reducers/articles.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ function articles(state = {}, action) {
       let response = action.response
       try {
         let nextHref = action.response.links.next.href
-        let embedded = JSON.stringify({ authors: 1, tags:1, categories:1 })
+        let embedded = JSON.stringify({ writters: 1, photographers:1, designers: 1, engineers:1, tags:1 })
```
Comment (hcchien) [inline src/reducers/articles.js]: add the heroImage embedded

File: src/utils/denormalize-articles.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    })
+  }
+  return rtn
+}
+
+export function denormalizeArticles(articleSlugs = [], entities = {}) {
+  let denormalizedArticles = []
+  // extract entities articles need
+  const { articles } = entities
+  const list = [ 'writters', 'photographers', 'designers', 'engineers', 'tags' ]
```
Comment (hcchien) [inline src/utils/denormalize-articles.js]: add heroImage

File: src/containers/Home.js [hunk 4]
```diff
@@ -1,76 +1,72 @@
...
+  fetchArticlesIfNeeded('feature', MAXRESULT, PAGE)
 }
 
+
 export default class Home extends Component {
   static fetchData({ store }) {
-    let params = [ 'hp-projects', 'review', 'feature' ]
-    return store.dispatch(loadMultiTaggedArticles(params))
+    // load tagged articles
+    store.dispatch(fetchArticlesIfNeeded('hp-projects', MAXRESULT, PAGE))
```
Comment (hcchien) [inline src/containers/Home.js]: we have a new field called featured.

**Comment by hcchien** [comment]
lgtm

File: src/containers/Home.js (hunk 4 above)
Comment (nickhsine) [inline src/containers/Home.js]: new tag?

**Pull request merged**